src/client.rs

```diff
@@ -1488,4 +1471,70 @@ mod test {
 
         assert_eq!(description, expected_description);
     }
+
+    #[test]
+    fn update_fails_if_cannot_write_to_repo() {
+        block_on(async {
+            let remote = EphemeralRepository::<Json>::new();
+
+            // First, create the metadata.
+            let _ = RepoBuilder::new(&remote)
+                .root_keys(vec![&KEYS[0]])
+                .targets_keys(vec![&KEYS[0]])
+                .snapshot_keys(vec![&KEYS[0]])
+                .timestamp_keys(vec![&KEYS[0]])
+                .with_root_builder(|bld| {
+                    bld.expires(Utc.ymd(2038, 1, 1).and_hms(0, 0, 0))
+                        .root_key(KEYS[0].public().clone())
+                        .snapshot_key(KEYS[0].public().clone())
+                        .targets_key(KEYS[0].public().clone())
+                        .timestamp_key(KEYS[0].public().clone())
+                })
+                .targets_version(1)
+                .commit()
+                .await
+                .unwrap();
+
+            // Now, make sure that the local metadata got version 1.
+            let local = ErrorRepository::new(EphemeralRepository::new());
+            let mut client = Client::with_trusted_root_keys(
+                Config::default(),
+                &MetadataVersion::Number(1),
+                1,
+                once(&KEYS[0].public().clone()),
+                &local,
+                &remote,
+            )
+            .await
+            .unwrap();
+
+            // The first update should succeed.
+            assert_matches!(client.update().await, Ok(true));
+
+            // Publish new metadata.
+            let _ = RepoBuilder::new(&remote)
+                .root_keys(vec![&KEYS[0]])
+                .targets_keys(vec![&KEYS[0]])
+                .snapshot_keys(vec![&KEYS[0]])
+                .timestamp_keys(vec![&KEYS[0]])
+                .with_root_builder(|bld| {
+                    bld.expires(Utc.ymd(2038, 1, 1).and_hms(0, 0, 0))
+                        .root_key(KEYS[0].public().clone())
+                        .snapshot_key(KEYS[0].public().clone())
+                        .targets_key(KEYS[0].public().clone())
+                        .timestamp_key(KEYS[0].public().clone())
+                })
+                .targets_version(2)
+                .snapshot_version(2)
+                .timestamp_version(2)
+                .commit()
+                .await
+                .unwrap();
+
+            local.fail_stores(true);
+
+            // The second update should fail.
```

Comment:
Should an immediate third/fourth/fifth/... update fail?  As-is, this would eventually succeed (though with false instead of the expected true result).